### ISSUE TITLE
Function to get frame_max and heartbeat from connection_state

### ISSUE THREAD

**Issue by pinepain** [alanxz/rabbitmq-c#243]
There are already `int amqp_get_channel_max(amqp_connection_state_t state)` to get `channel_max`, 
and it will be a good idea to defer `frame_max` and `heartbeat` from user-space code?

I need to have such public functions to get actual `frame_max` and `hearbeat` in [php-amqp](https://github.com/pdezwart/php-amqp).

Can you specify any estimates when you'll be rolling next release and will you include such features (very trivial one) in it if I provide PR?

**Comment by alanxz** [comment]
I would accept patches adding these features (if I don't get to it myself first). Out of curiosity: what would you use these two parameters for?

I could probably do a release within a week if the need was there.

**Comment by pinepain** [comment]
When RabbitMQ config values `channel_max`, `frame_max`, `heartbeats` set and their value are lower than client tries to negotiate, then broker send in response actual lower values.

TL;DR;

The whole issue noted in RabbitMQ errata [section 10 Special case value for channel-max and frame-max](http://www.rabbitmq.com/amqp-0-9-1-errata.html#section_10).

Here is what I found:

Function `amqp_tune_connection` called twice, in [amqp_connection.c#L80](https://github.com/alanxz/rabbitmq-c/blob/bfd8cc9129da46136691a88c06d505b78f0043ae/librabbitmq/amqp_connection.c#L80) (initializing basic values and memory allocation for 0 channel) and in  [amqp_socket.c#L1292](https://github.com/alanxz/rabbitmq-c/blob/bfd8cc9129da46136691a88c06d505b78f0043ae/librabbitmq/amqp_socket.c#L1292) when doing `amqp_login_inner`.

I've added `  printf("amqp_tune_connection: channel_max=%d, frame_max=%d, heartbeat=%d\n", channel_max, frame_max, heartbeat);` in `amqp_tune_connection` function (not the best way to debug, but it works this time for me), so when I tried to connect with `channel_max=256, frame_max=131072, heartbeats=600000` when server has such values (they just set to these values for testing purposes:
```
   {heartbeat, 6001},
   {frame_max, 13107},
   {channel_max, 2}
```
during connection I got:
```
amqp_tune_connection: channel_max=0, frame_max=65536, heartbeat=0
amqp_tune_connection: channel_max=2, frame_max=13107, heartbeat=6001
```

So we can't rely on requested `channel_max`, `frame_max` and `heartbeats` values in the userspace code.

**Comment by alanxz** [comment]
What do you use these values for in the userspace code?

**Comment by pinepain** [comment]
For informational purposes only, so users can get what is going on on their side.

**Comment by alanxz** [comment]
Release v0.6.0 is live.
